pkg/collections/transientdata/storeprovider/store/cache/cache.go

```diff
@@ -0,0 +1,157 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package cache
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/bluele/gcache"
+	"github.com/hyperledger/fabric/common/flogging"
+	"github.com/trustbloc/fabric-peer-ext/pkg/collections/transientdata/storeprovider/store/api"
+	"github.com/trustbloc/fabric-peer-ext/pkg/config"
+)
+
+var logger = flogging.MustGetLogger("memtransientdatastore")
+
+// Cache is an in-memory key-value cache
+type Cache struct {
+	cache   gcache.Cache
+	ticker  *time.Ticker
+	dbstore transientDB
+}
+
+// transientDB - an interface for persisting and retrieving keys
+type transientDB interface {
+	AddKey(api.Key, *api.Value) error
+	DeleteExpiredKeys() error
+	GetKey(key api.Key) (*api.Value, error)
+	Close()
+}
+
+// New return a new in-memory key-value cache
+func New(size int, dbstore transientDB) *Cache {
+	c := &Cache{
+		ticker:  time.NewTicker(config.GetTransientDataExpiredIntervalTime()),
+		dbstore: dbstore,
+	}
+
+	c.cache = gcache.New(size).
+		LoaderExpireFunc(c.loadFromDB).
+		EvictedFunc(c.storeToDB).
+		ARC().Build()
+
+	// cleanup expired data in db
+	go c.periodicPurge()
+
+	return c
+}
+
+// Close closes the cache
+func (c *Cache) Close() {
+	c.cache.Purge()
+	c.ticker.Stop()
+}
+
+// Put adds the transient value for the given key.
+// Returns the previous value (if any)
```

Comment:
Comments say the function returns prev value, but the code doesn't have any return.

Reply:
Fixed